### DISCUSSION TITLE
Vue Components Documentation

### DISCUSSION THREAD

**Discussion by jheslop25** [area17/twill#1224] [General]
Hey guys, feel free to call me horrible names if I've blatantly missed the docs for available vue components, but I can't find them to save my life. Am I correct that they aren't documented yet? 

Can someone point me to a list at least? 

Thanks!

**Comment by pboivin** [comment]
Hi @jheslop25, I think this is a really good question actually, no name calling today :)

So the short answer is no, there's no proper documentation for the Vue components. You can find them in this section of the codebase: https://github.com/area17/twill/tree/2.x/frontend/js/components

And you can find examples in Twill's code for specific components by searching with the `a17-` prefix. For example, if you want to find some use case of `Checkbox.vue`, you can search for `a17-checkbox`.

**Comment by jheslop25** [comment]
@pboivin Phew, I was getting worried that I was going blind (not the first time). Thanks for the quick reply though and giving as much info as you could! So far I'm really happy with Twill, the only criticism I have is that there are some undocumented features. The docs you do have are pretty great though and I deeply appreciate the development style!

**Comment by pboivin** [comment]
It's great to hear! I can say that I personally agree — 100% — that the docs are not in their best shape. To be honest, this has to be the # 1 negative feedback we get across all channels. But we're actively working on it and... turns out it's quite difficult to write good docs, so it's a slow process :)

One more thing I can share with you is a short guide to setup and develop custom Vue components for Twill: https://gist.github.com/pboivin/19d49cbb3c0a6c5c06469117597cc0f7

**Comment by jheslop25** [comment]
@pboivin LOL yeah I think every developer in history has found out the hard way that docs are incredibly hard! Although you guys seem to make up for it pretty well by being available and responsive on github. 

I'll have a look through that guide, it looks like exactly what I need! Thanks!

**Comment by ifox** [comment]
Hi @jheslop25, join us on Discord if you'd like! We're actively discussing all sorts of  ideas and issues with the community there: https://discord.link/twill
